platforms/chibios/boards/GENERIC_STM32_F407XE/configs/config.h

```diff
@@ -21,3 +21,13 @@
 #ifndef EARLY_INIT_PERFORM_BOOTLOADER_JUMP
 #    define EARLY_INIT_PERFORM_BOOTLOADER_JUMP TRUE
 #endif
+
+#ifdef WEAR_LEVELING_EMBEDDED_FLASH
+#    ifndef WEAR_LEVELING_EFL_FIRST_SECTOR
+#        ifdef BOOTLOADER_TINYUF2
+#            define WEAR_LEVELING_EFL_FIRST_SECTOR 3
+#        else
+#            define WEAR_LEVELING_EFL_FIRST_SECTOR 1
+#        endif
+#    endif
+#endif
```

Comment:
This won't work without also adding `platforms/chibios/boards/common/ld/STM32F407xE.ld` (and `platforms/chibios/boards/common/ld/STM32F407xE_tinyuf2.ld` for the tinyuf2 case).  Currently QMK will pick up `lib/chibios/os/common/startup/ARMCMx/compilers/GCC/ld/STM32F407xE.ld` for the stm32-dfu case, which does not reserve the second 16K flash sector for the EEPROM backing store.

It could be possible to take advantage of the recently added `stm32f4xx_common.ld` and `stm32f4xx_tinyuf2_common.ld` to simplify the ld files, but then we would lose support for the CCM SRAM (`ram4`) and BCKP SRAM (`ram5`) regions unless we modify the common file to support those regions too.

Reply:
Ah, okay.  If that's the case, I'm picking up a few weact boards to verify on hardware.

And f405 doesn't use the common config, so might be worth messing with that stuff too.

Thanks!

Reply:
Actually I have the WeAct STM32F405RGT6 and STM32F446RET6 boards (got them basically for the shipping cost during the recent sale; skipped the STM32F412RET6 variant, but it's not yet supported in QMK anyway).  As for F407, apparently WeAct does not offer any boards with that chip.

Reply:
Well, I do have the f405, and that's what started this whole journey.  And I've been considering the f446, so this is mostly just an excuse to pick it up.  (and a g474).  And yeah, no weact f407 boards, sadly.